core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/DefaultQuickPulsePingSender.java

```diff
@@ -50,37 +51,23 @@
     private final TelemetryConfiguration configuration;
     private final HttpClient httpClient;
     private final QuickPulseNetworkHelper networkHelper = new QuickPulseNetworkHelper();
-    private final String pingPrefix;
-    private final String roleName;
+    private volatile String pingPrefix; // cached for performance
     private final String instanceName;
     private final String machineName;
     private final String quickPulseId;
     private long lastValidTransmission = 0;
+    private String roleName;
```

Comment:
this looks unused to me, always gets overwritten in the one method where it's read, so can just be local var in `getPingPrefix()`

Reply:
`public HttpPost buildPingRequest(Date currentDate, String address, String quickPulseId, String machineName, String roleName, String instanceName) {

        HttpPost request = buildRequest(currentDate, address);
        request.addHeader(QPS_ROLE_NAME_HEADER, roleName);
        request.addHeader(QPS_MACHINE_NAME_HEADER, machineName);
        request.addHeader(QPS_STREAM_ID_HEADER, quickPulseId);
        request.addHeader(QPS_INSTANCE_NAME_HEADER, instanceName);
        request.addHeader(QPS_INVARIANT_VERSION_HEADER, Integer.toString(QuickPulse.QP_INVARIANT_VERSION));
        return request;
    }`

it's used here too.  RoleName is in the header as well.

Reply:
doh I missed that, thx 👍